### DISCUSSION TITLE
What is the use of EXTRA_DEPENDS?

### DISCUSSION THREAD

**Discussion by jetm** [priv-kweihmann/meta-sca#8102] [Q&A] [answered]
Hi,

All the generated Go recipes have `EXTRA_DEPENDS` but the `go-gen` code has no other use:

https://github.com/priv-kweihmann/meta-sca/blob/3ba84c187f918b69f49c9fa88815fb412154f11c/scripts/go-gen#L399

**Comment by priv-kweihmann** [comment, marked as answer]
I introduced EXTRA_DEPENDS as variable that can be set manually.
Basically in a generated recipe DEPENDS is extracted automatically and enhanced by the manual settings from EXTRA_DEPENDS.

By this one can inject dependencies that can't be determined by the recipe generator. Plus in case of an update the settings of EXTRA_DEPENDS are taken over from the older recipe(s)

Does this answer your question?

**Comment by jetm** [comment]
For clarification to see if I understand, at the moment, `EXTRA_DEPENDS` only is treated as data that eventually could be reused to update generated recipes?

**Reply by priv-kweihmann** [reply]
Yes, exactly - It's meant as a way to preserve dependencies that have to be defined manually.
Classic example:

a recipe depends on some c library, which can't be seen by the recipe generator, so typically one would inject this dependency via `EXTRA_DEPENDS += "some-lib"`

With this it would be preserved with the next update of the generated recipe

**Reply by jetm** [reply]
Thank you!
